tensorboard/components/tf_runs_selector/tf-wbr-string.html

```diff
@@ -0,0 +1,75 @@
+<!--
+@license
+Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="../tf-imports/polymer.html" />
+
+<!--
+  tf-wbr-string safely renders a string, with <wbr> elements inserted
+  around select delimiters.
+-->
+<dom-module id="tf-wbr-string">
+  <template>
+    <!--
+      This ugly formatting is required to prevent spaces from slipping
+      into the HTML.
+    -->
+    <template is="dom-repeat" items="[[_parts]]" as="part"
+      >[[part.contents]]<template is="dom-if" if="[[!part.last]]"
+        ><wbr /></template
+    ></template>
+  </template>
+  <script>
+    Polymer({
+      is: 'tf-wbr-string',
+      properties: {
+        /** The value to render with word breaks. */
+        value: String,
+        /**
+         * Insert a word-break after each match of this regex.
+         *
+         * This `RegExp` object will not be mutated.
+         */
+        delimiterPattern: {
+          type: Object /* RegExp */,
+          value: /([\/=\-_,])/,
+        },
+        _parts: {
+          type: Array /* {contents: string, last: boolean}[] */,
+          computed: '_computeParts(value, delimiterPattern)',
+        },
+      },
+      _computeParts(value, delimiterPattern) {
+        const result = [];
+        if (value == null) {
+          value = '';
+        }
+        while (true) {
+          const idx = value.search(delimiterPattern);
+          if (idx === -1) {
+            result.push({contents: value, last: true});
```

Comment:
If value was empty (i.e. the last delimiter matched is at the end of the string), then we'll be pushing an extra `<wbr>` onto the end, which seems unnecessary.

Seems like a fix would just be to do `while (value)` instead?

Reply:
Either seems fine with me (they should have the same effect), but sure.

Reply:
Using `while (value)` doesn’t actually prevent the last-added part from
having `last: false` on an input like `"a/"`, and thus doesn’t prevent
the extra `<wbr>`. Given that the unnecessary `<wbr>`s do no harm, and
this code is clearly proving too complicated for my tiny brain, I’ve
gone the other way and just included the `<wbr>` after every part. Much
simpler. Everybody’s a winner.